### ISSUE TITLE
Multiple UserDetailsManager in VaadinWebSecurity for Hilla

### ISSUE THREAD

**Issue by egloffmark** [vaadin/hilla#1085]
**Is your feature request related to a problem? Please describe.**
I like to raise a feature request or asking for an example how I can use multiple `UserDetailsManager` or `UserDetailsService`with `VaadinWebSecurity` and Hilla. Background, I am creating an app with Vaadin Flow and Spring Boot + Security which will have a Frontend for Registered Customers and a Backend for the App Administration. Unfortunately the Data Sources for the users Frontend and Backend will not be the same.

I tried to instantiate two instances of VaadinWebSecurity but it seems not to work, because looking at the source code of Vaadin it is based on named Beans. Only one seems to be supported.

To follow an example to do it with plain Spring Security like ["Multiple Entry Points in Spring Security" ](https://www.baeldung.com/spring-security-multiple-entry-points), will not work because the base class `VaadinWebSecurity`does a lot of framework internal configurations also for Hilla.

**Describe the solution you'd like**
Split the VaadinWebSecurity in parts which can be independently used, overwritten and allows multiple instances. It should allow to add multiple Security Configurations and Security Filters and makes it transparent how it the Vaadin WebFlow and Hilla get configured.

**Describe alternatives you've considered**
Rewrite and create own Spring VaaDinWebSecurity class but this requires too much framework know how

**Additional context**
I am using Hilla 2, Spring Boot 3 and Spring Security 6. In Spring Security 6 the old Spring configuration class `WebSecurityConfigurerAdapter` is no longer supported. The solution should consider therefore Spring 5 and Spring 6

**Comment by cromoteca** [comment]
Thank you for reporting this: it looks like a valid use case. I don't think that this is a Hilla issue as the web security is shared with Flow. Some joint investigation is needed.

**Comment by egloffmark** [comment]
I agree, however I like to use Hilla for both frontend and backend and therefore looking forward that it get once supported from vaadin-flow and hilla perspective. Thank you

**Comment by tarekoraby** [comment]
@cromoteca, if this is not a Hilla issue, should the ticket be then moved to the Flow repo?

**Comment by egloffmark** [comment]
I opened this other ticket yes, however I assume after vaadin-flow got updated you also need to do some updates on hilla.
